java/mdsplus-api/src/main/java/mds/mdsip/Message.java

```diff
@@ -113,19 +120,20 @@ protected static final ByteBuffer readCompressedBuf(final ReadableByteChannel rb
 		rbc.read(unflattened_length);
 		final int bytes_to_read = unflattened_length.getInt(0) - Message.HEADER_SIZE;
 		try(CompressedChannel cc = new Message.CompressedChannel(rbc);){
-			return Message.readBuf(bytes_to_read, cc, listeners);
+			return Message.readBuf(bytes_to_read, cc, listeners, 0);
```

Comment:
What does the 0 do?

Reply:
this would trigger a timeout if the server stopped sending bytes. as it receiving followup bytes of a message the server should not pause unless something is wrong.. timeout of 0 means not timeout.. or non-blocking read